test/e2e/framework/util.go

```diff
@@ -1359,14 +1359,14 @@ func podNotPending(c clientset.Interface, podName, namespace string) wait.Condit
 // to terminate or if the pod terminated with an unexpected reason.
 func waitForPodTerminatedInNamespace(c clientset.Interface, podName, reason, namespace string) error {
 	return WaitForPodCondition(c, namespace, podName, "terminated due to deadline exceeded", PodStartTimeout, func(pod *v1.Pod) (bool, error) {
-		if pod.Status.Phase == v1.PodFailed {
-			if pod.Status.Reason == reason {
+		if pod.Status.Phase == v1.PodSucceeded || pod.Status.Phase == v1.PodFailed {
+			if reason == "" ||
+				strings.ToLower(strings.TrimSpace(pod.Status.Reason)) == strings.ToLower(strings.TrimSpace(reason)) {
```

Comment:
At least the original behavior of passing in "" as the reason implied that the caller wants to delete the pod and expects it to succeed.  If the pod failed, then the wait will timeout and the cause the test to fail.  But this change would cause the test to pass even if Pod deletion failed, which I don't think we want.  I can't think of a reason why the caller wouldn't care if the Pod failed.  It is not an expected condition and indicates there is a problem somewhere.